### PR TITLE
Issue 598 - Permissive value in filter

### PR DESCRIPTION
**Pull request by Marc-Andre-Rivet**
#598
Allow more permissive expression value (with whitespaces without escaping or quotes) in column filters.

A limitation is that if it starts with an operator's text, the operator will be picked up instead. e.g `le jour` is `<= jour`, not `"le jour"`.

Added a test case for permissive strings. 
Added changelog entry.

issue598

#597 
New `is blank` unary operator that returns true for `undefined`, `null` and `''`.

In both cases, the documentation will need to be updated.

### CODE REVIEW

File: src/dash-table/syntax-tree/lexeme/expression.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { ISyntaxTree } from 'core/syntax-tree/syntaxer';
 const FIELD_REGEX = /^{(([^{}\\]|\\.)+)}/;
 const STRING_REGEX = /^(('([^'\\]|\\.)*')|("([^"\\]|\\.)*")|(`([^`\\]|\\.)*`))/;
 const VALUE_REGEX = /^(([^\s'"`{}()\\]|\\.)+)(?:[\s)]|$)/;
+const PERMISSIVE_VALUE_REGEX = /^(([^'"`{}()\\]|\\.)+)(?:$)/;
```
Comment (alexcjohnson) [inline src/dash-table/syntax-tree/lexeme/expression.ts]:
```suggestion
const PERMISSIVE_VALUE_REGEX = /^(([^'"`{}()\\]|\\.)+)$/;
```
Minor, no point not capturing the `$` right?

File: src/dash-table/syntax-tree/lexeme/expression.ts [hunk 2]
```diff
@@ -55,17 +46,41 @@ export const stringExpression: IUnboundedLexeme = {
...
+        type: LexemeType.Expression
+    };
+};
+
+export const valueExpression = valueExpressionFactory(VALUE_REGEX);
+export const permissiveValueExpression = valueExpressionFactory(
+    PERMISSIVE_VALUE_REGEX,
+    (v: any) => typeof v === 'string' && v.indexOf(' ') !== -1 ?
+        `"${v}"` :
+        v
```
Comment (alexcjohnson) [inline src/dash-table/syntax-tree/lexeme/expression.ts]: Nice solution - esp. as it relates to the generated query string. ✨

File: tests/cypress/tests/unit/single_column_syntactic_tree_test.ts [hunk 3]
```diff
@@ -97,6 +97,14 @@ describe('Single Column Syntax Tree', () => {
         expect(tree.toQueryString()).to.equal('{a} = 1');
     });
 
+    it.only('can be permissive value expression', () => {
+        const tree = new SingleColumnSyntaxTree('Hello world', COLUMN_TEXT);
+
+        expect(tree.isValid).to.equal(true);
+        expect(tree.evaluate({ a: 'Hello world' })).to.equal(true);
+        expect(tree.evaluate({ a: 'Helloworld' })).to.equal(false);
+    });
```
Comment (alexcjohnson) [inline tests/cypress/tests/unit/single_column_syntactic_tree_test.ts]: Can you include a `toQueryString` test for permissive as well?

Comment (Marc-Andre-Rivet) [reply tests/cypress/tests/unit/single_column_syntactic_tree_test.ts]: Sure

**Pull request merged**